src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -3013,6 +3013,15 @@ module EstablishTypeDefinitionCores =
                     let kind = if hasMeasureAttr then TyparKind.Measure else TyparKind.Type
                     let ty, _ = TcTypeOrMeasureAndRecover (Some kind) cenv NoNewTypars checkConstraints ItemOccurence.UseInType WarnOnIWSAM.No envinner tpenv rhsType
 
+                    match ty with
```

Comment:
This most likely is bad code, please let me know how I can properly check for the `AutoOpenAttribute` here.

Reply:
Changed it to the following. Does that make sense?
```
if tcref.CompiledRepresentationForNamedType.FullName = g.attrib_AutoOpenAttribute.TypeRef.FullName then
                            
    warning(Error(FSComp.SR.chkAutoOpenAttributeInTypeAbbrev(), tycon.Id.idRange))
```